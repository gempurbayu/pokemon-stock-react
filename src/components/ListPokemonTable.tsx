/* eslint-disable react-hooks/exhaustive-deps */
import React, { useEffect } from 'react'
import { useDispatch, useSelector } from 'react-redux'
import { RootState } from '../store';
import { setDataPokemon } from '../store/actions';


const ListPokemonTable = () => {

    const dispatch = useDispatch();
    useEffect(() => {
        dispatch(setDataPokemon() as any);
      },[]);
    
    let pokemons = useSelector((state: RootState)=> state.pokemons.pokemon);

  return (
    <div>
        <table className='table-auto mt-6 w-full font-nunito'>
            <thead>
                <tr className='h-12 border-b-slate-800 border-b-[1px]'>
                <th  className='text-start w-6/12'>Nama</th>
                <th  className='text-end  w-6/12'>Stok</th>
                </tr>
            </thead>
            <tbody>
                {pokemons.map((item:any) => 
                    <tr className='h-12 border-b-slate-200 border-b-[1px]'>
                    <td className='text-start text-inti font-bold'>{item.name}</td>
                    <td className='text-end font-bold'>{item.stock}</td>
                    </tr>
                )}
            </tbody>
            </table>
    </div>
  )
}

export default ListPokemonTable