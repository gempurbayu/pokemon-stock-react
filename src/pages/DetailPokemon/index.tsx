import React, { useState } from 'react'
import { useParams } from 'react-router-dom'
import Header from '../../components/Header'
import UpdateStokModal from '../../components/UpdateStokModal';

const DetailPokemon = () => {
    let { id } = useParams();
    
    let [open, setOpenModal] = useState(false);

    const handleModal = () => {
        setOpenModal(true);
    }

    const closeModal = () => {
        setOpenModal(false);
    }

  return (
    <div className=''>
        <Header/>
        <div className='w-full h-full px-[4%]'>
            {/* Title Page */}
            <div className='section2 w-full mt-6 flex flex-col font-rubik h-140 py-4'>
                <h1 className='text-[28px] font-bold mb-6 text-basecolor'>Pikachu</h1>
                <button className='border-gray-200 border-2 rounded-[4px] py-0.5 text-sm w-[105px] h-10 font-bold text-inti bg-gray-100 shadow-sm'
                        onClick={handleModal}
                >Update Stok</button>
            </div>
            <div className='stok-info flex flex-col w-full text-start mt-4 mb-[40px]'>
                <p className='text-sm'>Sisa stok</p>
                <p className='text-3xl'>10 pcs</p>
            </div>
            <div className='riwayat w-full text-start flex flex-col'>
                <div className='mb-4'>
                    <p className='text-[16px] text-basecolor font-bold'>Riwayat stok</p>
                    <p className='text-sm text-basecolor '>Satuan stok dalam pcs</p>
                </div>
                <div className='w-full'>
                    <table className='w-full'>
                        <tr className='w-full font-bold text-basecolor h-12 border-b-2 border-gray-500'>
                            <td className='w-6/12'>2 Apr 2021</td>
                            <td className='w-3/12 text-right'>Jmlh</td>
                            <td className='w-3/12 text-right'>Stok</td>
                        </tr>
                        <tr className='w-full text-basecolor h-[88px] my-auto border-b-2'>
                            <td className='w-6/12'>
                                <div className='desc w-full flex flex-col'>
                                    <span className='text-[12px] font-bold'>08:00</span>
                                    <span className='text-sm font-bold text-inti'>Update stok</span>
                                    <span className='text-[12px]'>"Stok awal"</span>
                                </div>
                            </td>
                            <td className='w-3/12 text-right text-inti text-sm'>+10</td>
                            <td className='w-3/12 text-right font-bold text-basecolor'>{id}</td>
                        </tr>
                    </table>
                </div>
            </div>
        </div>
        <UpdateStokModal open={open} close={closeModal}/>
       
    </div>
  )
}

export default DetailPokemon