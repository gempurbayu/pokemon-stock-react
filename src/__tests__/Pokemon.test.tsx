import React from 'react';
import { render, screen, waitFor } from '@testing-library/react';
import { Provider } from 'react-redux';
import store from '../store';
import ListPokemonTable from '../components/ListPokemonTable';
import { BrowserRouter as Router, Route, Routes } from 'react-router-dom';
import DetailPokemon from '../pages/DetailPokemon';
import renderer from 'react-test-renderer';

describe("PokemonList Component", () => {
  it("should render pokemon same pokemon name with search", async () =>{
    render(
      <Provider store={store}>
        <Router>
          <ListPokemonTable search='ivysaur'/>
        </Router>
      </Provider>
    )
    await waitFor(() => {
      screen.getByText("ivysaur")
    })
  })
})

describe("Pokemon Detail", () => {
    it("should render pokemon names when api responds", async () =>{
        const component = renderer.create(
            <Provider store={store}>
            <Router>
                <Routes>
                <Route path='/pokemon/4' element={<DetailPokemon/>}>
                </Route>
                </Routes>
            </Router>
            </Provider>
         );
          let tree = component.toJSON();
          expect(tree).toMatchSnapshot();

          tree = component.toJSON();
          expect(tree).toMatchInlineSnapshot(`null`);
   })
  })
  
