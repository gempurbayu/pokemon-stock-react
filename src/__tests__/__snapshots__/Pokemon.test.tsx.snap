// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pokemon Detail should render pokemon names when api responds 1`] = `null`;
